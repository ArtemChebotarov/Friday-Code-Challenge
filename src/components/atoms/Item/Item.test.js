import React from "react";
import { cleanup } from "@testing-library/react";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { configure, shallow } from "enzyme";
import { Item } from "../index";
import { shallowToJson } from "enzyme-to-json";

configure({ adapter: new Adapter() });
afterEach(cleanup);

describe("Item test", () => {
  it("Test hover effect", () => {
    const mockCallBack = jest.fn();

    const item = shallow(
      <Item
        selected={false}
        id={0}
        setSelected={mockCallBack}
        content={"Test"}
      />
    );
    expect(shallowToJson(item)).toMatchSnapshot();

    item.prop("onMouseEnter")();
    expect(shallowToJson(item)).toMatchSnapshot();

    item.prop("onMouseLeave")();
    expect(shallowToJson(item)).toMatchSnapshot();
  });
});
