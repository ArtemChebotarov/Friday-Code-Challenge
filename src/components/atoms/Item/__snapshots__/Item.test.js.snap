// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item test Test hover effect 1`] = `
<div
  className="item"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="item--indicator"
  >
    <img
      alt="indicator"
      src="orange-empty-dot.svg"
    />
  </div>
  <p
    className="normal"
  >
    Test
  </p>
</div>
`;

exports[`Item test Test hover effect 2`] = `
<div
  className="item"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="item--indicator"
  >
    <img
      alt="indicator"
      src="orange-empty-dot.svg"
    />
  </div>
  <p
    className="hovered"
  >
    Test
  </p>
</div>
`;

exports[`Item test Test hover effect 3`] = `
<div
  className="item"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="item--indicator"
  >
    <img
      alt="indicator"
      src="orange-empty-dot.svg"
    />
  </div>
  <p
    className="normal"
  >
    Test
  </p>
</div>
`;
